layers.go

```diff
@@ -2392,6 +2397,12 @@ func (r *layerStore) ApplyDiffFromStagingDirectory(id, stagingDirectory string,
 	layer.UncompressedDigest = diffOutput.UncompressedDigest
 	layer.UncompressedSize = diffOutput.Size
 	layer.Metadata = diffOutput.Metadata
+	if expectedDiffID != "" {
+		if layer.Flags == nil {
+			layer.Flags = make(map[string]interface{})
+		}
+		layer.Flags[ExpectedLayerDiffIDFlag] = expectedDiffID
```

Comment:
If `diffOutput.UncompressedDigest` isn't empty, its value is already being saved in the layer's `UncompressedDigest` field.  How does storing a duplicate of it as a flag change anything?

Reply:
ApplyDiffFromStagingDirectory replaces `diffOutput.UncompressedDigest` with the TOC Digest.

And we are abusing `UncompressedDigest` since the TOC Digest was validated, while the layer uncompressed digest was not as we have not pulled the whole layer